### ISSUE TITLE
Custom data operations change the state of the data frame without any easy way to revert back

### ISSUE THREAD

**Issue by armish** [hammerlab/celldom#75]
When I do this type of filtering:

<a href="https://cl.ly/7047d0f48e62" target="_blank"><img src="https://d2ddoduugvun08.cloudfront.net/items/032L0r2X2S2k1G393n1G/Image%202019-04-19%20at%203.36.42%20PM.png" style="display: block;height: auto;width: 100%;"/></a>

the `df`  changes permanently without an easy way to undo the change. Restarting the app solves the issue but `df` should ideally be `immutable` to prevent irreversible data filtering.

**Comment by eric-czech** [comment]
Thanks @armish , it should be possible to undo the filtering by removing any code in that text area and then hitting apply again.  Just to be sure, that wasn't working right?

I recently overhauled a lot of the code for the new virtualized tables and I probably messed something up related to the filtering.

**Comment by armish** [comment]
That is correct -- removing the code and applying the changes didn't help. Not the most urgent issue, though. Just wanted to have this documented.
